tests/raftstore_cases/test_merge.rs

```diff
@@ -449,41 +448,59 @@ fn test_merge_approximate_size_and_keys() {
     let region = pd_client.get_region(b"").unwrap();
 
     cluster.must_split(&region, &middle_key);
+    // make sure split check is invoked so size and keys are updated.
     thread::sleep(Duration::from_secs(1));
 
     let left = pd_client.get_region(b"").unwrap();
     let right = pd_client.get_region(&max_key).unwrap();
-
     assert_ne!(left, right);
+
+    // make sure all peer's approximate size is not None.
+    cluster.must_transfer_leader(right.get_id(), right.get_peers()[0].clone());
+    thread::sleep(Duration::from_secs(1));
+    cluster.must_transfer_leader(right.get_id(), right.get_peers()[1].clone());
+    thread::sleep(Duration::from_secs(1));
+    cluster.must_transfer_leader(right.get_id(), right.get_peers()[2].clone());
+    thread::sleep(Duration::from_secs(1));
+
     let size = pd_client
-        .get_region_approximate_size(left.get_id())
-        .unwrap()
-        + pd_client
-            .get_region_approximate_size(right.get_id())
-            .unwrap();
+        .get_region_approximate_size(right.get_id())
+        .unwrap();
     assert_ne!(size, 0);
     let keys = pd_client
-        .get_region_approximate_keys(left.get_id())
-        .unwrap()
-        + pd_client
-            .get_region_approximate_keys(right.get_id())
-            .unwrap();
+        .get_region_approximate_keys(right.get_id())
+        .unwrap();
     assert_ne!(keys, 0);
 
     pd_client.must_merge(left.get_id(), right.get_id());
-    thread::sleep(Duration::from_secs(1));
+    // make sure split check is invoked so size and keys are updated.
+    thread::sleep(Duration::from_secs(2));
```

Comment:
Why sleep guarantees that split check is invoked?

Reply:
split check every 100ms